googlesource/googlesource.go

```diff
@@ -90,35 +119,76 @@ func (s *Source) ReadGoogleRow(rowInput chan readRowInput, responseCh chan sdk.R
 				break
 			}
 			if err != nil {
-				sdk.Logger(s.Ctx).Error().Str("err", err.Error()).Msg("error while iterating")
+				sdk.Logger(s.ctx).Error().Str("err", err.Error()).Msg("error while iterating")
 				return err
 			}
 
-			offset++
-			position := Position{
-				TableID: tableID,
-				Offset:  offset,
+			data := make(sdk.StructuredData)
+			var key Key
+
+			for i, r := range row {
+				// handle dates
+				if schema[i].Type == bigquery.TimestampFieldType {
+					dateR := fmt.Sprintf("%v", r)
+					dateLocal, err := time.Parse("2006-01-02 15:04:05.999999 -0700 MST", dateR)
```

Comment:
Why do we have this time format?

Reply:
The time we get here is of a different format  (Golang's format) which we see on BIgquery. This is to convert it into the Bigquery provided format.